lib/connection.js

```diff
@@ -109,15 +110,38 @@ class Connection extends EventEmitter {
   }
 
   // Send a handshake packet over the connection
-  //   appName - name of an application to connect to
+  //   app - string or object, application to connect to as 'name' or
+  //         'name@version' or { name, version }
   //   login - user name (optional)
   //   password - user password (optional)
   //   callback - callback function to invoke after the handshake is completed
   //
-  handshake(appName, login, password, callback) {
-    const packet = login && password ?
-      this._createPacket('handshake', appName, 'login', [login, password]) :
-      this._createPacket('handshake', appName);
+  handshake(app, login, password, callback) {
+    let name, version;
+    if (typeof app === 'string') {
+      [name, version] = app.split('@');
+      if (version && !semver.valid(version)) {
```

Comment:
Shouldn't this method allow providing a range instead of exact version, if you enabled support for the ranges on the server side (`_getApplication` method in `lib/server.js`)?